CMakeLists.txt

```diff
@@ -7,63 +7,56 @@
 # ======================================================================
 
 
-# use cmake 3.14 or later
-cmake_minimum_required (VERSION 3.12...3.14 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.20 FATAL_ERROR)
 
-project(larsimdnn LANGUAGES CXX)
-
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
-
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules $ENV{ART_DIR}/Modules)
+find_package(cetmodules REQUIRED)
+project(larsimdnn VERSION 09.00.00 LANGUAGES CXX)
 
 include(CetCMakeEnv)
 cet_cmake_env()
 
 cet_set_compiler_flags(DIAGS CAUTIOUS
   WERROR
   NO_UNDEFINED
-  EXTRA_FLAGS -pedantic -Wno-unused-local-typedefs
+  EXTRA_FLAGS -Wno-unused-local-typedefs
 )
 
-cet_report_compiler_flags()
+cet_report_compiler_flags(REPORT_THRESHOLD VERBOSE)
+
+find_package(larfinder REQUIRED)
+
+find_package(larsim REQUIRED EXPORT)
+find_package(larevt REQUIRED EXPORT)
+find_package(lardata REQUIRED EXPORT)
+find_package(larcore REQUIRED EXPORT)
+find_package(lardataobj REQUIRED EXPORT)
+find_package(larcorealg REQUIRED EXPORT)
+find_package(larcoreobj REQUIRED EXPORT)
 
-# these are minimum required versions, not the actual product versions
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( larcore )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( lardata )
-find_ups_product( larevt )
-find_ups_product( larsim )
-find_ups_product( nutools )
-find_ups_product( nug4 )
-find_ups_product( nurandom )
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( postgresql )
-find_ups_product( range )
-find_ups_product( eigen )
+find_package(nusimdata REQUIRED EXPORT)
+find_package(nurandom REQUIRED EXPORT)
+
+find_package(art_root_io REQUIRED EXPORT)
+find_package(art REQUIRED EXPORT)
+find_package(canvas REQUIRED EXPORT)
+find_package(messagefacility REQUIRED EXPORT)
+find_package(cetlib REQUIRED EXPORT)
+find_package(cetlib_except REQUIRED EXPORT)
+
+find_package(CLHEP COMPONENTS Random REQUIRED EXPORT)
+find_package(ROOT COMPONENTS Core Hist Tree REQUIRED EXPORT)
+find_package(TBB REQUIRED EXPORT)
+find_package(TensorFlow 2.6.0 QUIET EXPORT)
+find_package(Threads REQUIRED EXPORT)
```

Comment:
I'm a bit nervous about all these explicit `find_pacakge()` calls before we have any code to use them. I'd leave them out and only put in what we need, when we need it.

Reply:
leaving tensorflow in place since that's the point of this PR.  We'll likely need to restore some of these.  Happy to wait and see.